### ISSUE TITLE
How to evaluate bbox and segm with Cascade Mask R-CNN?

### ISSUE THREAD

**Issue by TuanNguyenSKKU** [open-mmlab/mmdetection#3548]
I can print the results of Mask R-CNN with the command below.
`python tools/test.py configs/mask_rcnn/mask_rcnn_r101_fpn_1x_coco.py work_dirs/mask_rcnn_r101_fpn_1x_coco/latest.pth --eval bbox segm --out results.pkl`

However, the error happens for Cascade Mask R-CNN (see photo). 
Please help me to deal with this.
Thank you.
![image](https://user-images.githubusercontent.com/55343855/90112880-8af92d80-dd8b-11ea-8f61-2203aafe7717.png)

**Comment by ZwwWayne** [comment]
You need to check your config. It seems that you config changes the `num_classes` from 80 to 4 in some layers thuse the model weights are not correctly loaded.

**Comment by TuanNguyenSKKU** [comment]
Thank you. 
I changed num_classes from 80 to 3 for my custom dataset (for both Mask R-CNN and Cascade Mask R-CNN) and I keep the default for the remaining. 
But the problem is that everything was fine for Mask R-CNN but it was an error for Cascade Mask R-CNN.
This is the result of Mask R-CNN after running the above command:
![image](https://user-images.githubusercontent.com/55343855/90131835-bd188880-dda7-11ea-961c-8ebd12ff6d90.png)

**Comment by TuanNguyenSKKU** [comment]
This is a file that I used to create pretrained_weights of Cascade Mask R-CNN. I hope that can help you know what is happening.

[import torch

pretrained_weights = torch.load('checkpoints/cascade_mask_rcnn_r50_fpn_1x_coco_20200203-9d4dcb24.pth')

num_class = 3
pretrained_weights['state_dict']['roi_head.bbox_head.0.fc_cls.weight'].resize_(num_class+1, 1024)
pretrained_weights['state_dict']['roi_head.bbox_head.0.fc_cls.bias'].resize_(num_class+1)
pretrained_weights['state_dict']['roi_head.bbox_head.1.fc_cls.weight'].resize_(num_class+1, 1024)
pretrained_weights['state_dict']['roi_head.bbox_head.1.fc_cls.bias'].resize_(num_class+1)
pretrained_weights['state_dict']['roi_head.bbox_head.2.fc_cls.weight'].resize_(num_class+1, 1024)
pretrained_weights['state_dict']['roi_head.bbox_head.2.fc_cls.bias'].resize_(num_class+1)

pretrained_weights['state_dict']['roi_head.mask_head.0.conv_logits.weight'].resize_(num_class, 256, 1, 1)
pretrained_weights['state_dict']['roi_head.mask_head.0.conv_logits.bias'].resize_(num_class)

pretrained_weights['state_dict']['roi_head.mask_head.1.conv_logits.weight'].resize_(num_class, 256, 1, 1)
pretrained_weights['state_dict']['roi_head.mask_head.1.conv_logits.bias'].resize_(num_class)

pretrained_weights['state_dict']['roi_head.mask_head.2.conv_logits.weight'].resize_(num_class, 256, 1, 1)
pretrained_weights['state_dict']['roi_head.mask_head.2.conv_logits.bias'].resize_(num_class)

torch.save(pretrained_weights, "cascade_mask_rcnn_r50_fpn_1x_%d.pth"%num_class)](url)
